### PR TITLE
Implementation derivative support for MatrixSolve

### PR DESCRIPTION
**Pull request by tjstienstra**
#### References to other Issues or PRs

Implements #24864

#### Brief description of what is fixed or changed
Implement the support of taking a derivative of a `MatrixSolve` node.

#### Release Notes

* codegen
  * Implemented diff support for `MatrixSolve`.

### CODE REVIEW

File: sympy/codegen/tests/test_matrix_nodes.py [hunk 1]
```diff
@@ -1,9 +1,51 @@
...
+    assert simplify(x_lu.diff(q) - dxdq_lu) == zeros(2, 1)
+    # dxdq_ms is the MatrixSolve equivalent of dxdq_lu
+    dxdq_ms = MatrixSolve(A, b.diff(q) - A.diff(q) * MatrixSolve(A, b))
+    assert MatrixSolve(A, b).diff(q) == dxdq_ms
+
+
+def test_matrix_solve_derivative_numpy():
+    np = import_module('numpy')
+    if not np:
+        skip("numpy not installed.")
```
Comment (brocksam) [inline sympy/codegen/tests/test_matrix_nodes.py:36]: Is this considered the best practice way to run tests with optional dependencies?

Comment (brocksam) [reply sympy/codegen/tests/test_matrix_nodes.py:36]: Have just checked the `lambdify` tests that have lots of optional dependency reliances and they do indeed do it exactly this way so I'm happy.

**Review by brocksam — COMMENTED** [review]
This looks good to me. And great addition of really useful functionality! Thanks!

**Review by brocksam — COMMENTED** [review]
We need the extra blank lines at end of file removed for the code quality check to pass though.

**Comment by moorepants** [comment]
Nice addition! LGTM too.

**Pull request merged**